custom_components/elasticsearch/es_privilege_check.py

```diff
@@ -33,10 +33,21 @@ def __init__(self, es_gateway: ElasticsearchGateway, config: dict = {}):
     async def enforce_privileges(self, config: dict = None):
         """Ensure client is configured with properly authorized credentials."""
         LOGGER.debug("Starting privilege enforcement")
-        result = await self.check_privileges(config if config else self.config)
-        if not result.has_all_requested:
-            LOGGER.debug("Required privileges are missing.")
-            raise InsufficientPrivileges()
+
+        # TODO: Figure out why this is needed
+        resultantConfig = config or self.config
+
+        if (  # is_authenticated
+            CONF_USERNAME in resultantConfig
+            and resultantConfig.get(CONF_USERNAME) is not None
+            or CONF_API_KEY in resultantConfig
+            and resultantConfig.get(CONF_API_KEY) is not None
+        ):
```

Comment:
I believe this can be simplified:

```suggestion
        if (  # is_authenticated
            resultantConfig.get(CONF_USERNAME, None) is not None
            or resultantConfig.get(CONF_API_KEY, None) is not None
        ):
```